### ISSUE TITLE
Change concrete *log.Logger to mesh.Logger interface

### ISSUE THREAD

**Issue by peterbourgon** [weaveworks/mesh#37]
Proposed interface

```go
type Logger interface {
    Logf(format string, args ...interface{})
}
```

**Comment by bboreham** [comment]
Compare https://github.com/Sirupsen/logrus/blob/master/logrus.go#L97

(not claiming I can see an easy middle ground)

See also https://github.com/weaveworks/weave/commit/c38dcb1d73349eaf5336d6541552e1edb15ff7db

**Comment by peterbourgon** [comment]
Yeah, there's a lot to consider. I'm definitely against an interface with more than 1 method, see [this presentation from a Go kit collaborator](https://go-talks.appspot.com/github.com/ChrisHines/talks/structured-logging/structured-logging.slide#1) for a good recap of my journey with logging in Go kit. And sorry to make you use that logLogger.
